### PR TITLE
swagger recording tests fixes

### PR DESCRIPTION
**Pull request by matthchr**

### CODE REVIEW

File: v2/tools/generator/internal/codegen/pipeline/determine_resource_ownership.go [hunk 1]
```diff
@@ -83,18 +87,24 @@ func findChildren(rt *astmodel.ResourceType, resourceName astmodel.TypeName, oth
...
+		otherURI := canonicalizeURI(other.ARMURI())
+
 		if strings.HasPrefix(otherURI, myPrefix) {
 			// now, accept it only if it contains two '/' exactly:
 			// so that the string is of the form:
 			//     {prefix}/resourceType/resourceName
 			// and not a grandchild resource:
 			//     {prefix}/resourceType/resourceName/anotherResourceType/anotherResourceName
-			withoutPrefix := otherURI[len(myPrefix)-1:]
+			withoutPrefix := otherURI[len(myPrefix)-1:] // TODO: Case should be ignored here
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/codegen/pipeline/determine_resource_ownership.go:107]: Should this TODO be addressed now?

File: v2/internal/controllers/crd_eventgrid_domain_test.go [hunk 2]
```diff
@@ -101,31 +100,33 @@ func Test_EventGrid_Domain(t *testing.T) {
...
 			Destination: &eventgrid.EventSubscriptionDestination{
 				StorageQueue: &eventgrid.StorageQueueEventSubscriptionDestination{
-					EndpointType: &endpointType,
-					Properties: &eventgrid.StorageQueueEventSubscriptionDestinationProperties{
-						ResourceReference: acctReference,
-						QueueName:         &queue.Name,
-					},
+					EndpointType: endpointType, // TODO[donotmerge]: This should be a ptr but isn't, see https://github.com/Azure/azure-service-operator/issues/2619
+					// TODO[donotmerge]: These properties used to be in a "Properties" property but are flattened
+					// TODO[donotmerge]: in the Swagger branch
```
Comment (theunrepentantgeek) [inline v2/internal/controllers/crd_eventgrid_domain_test.go:123]: Address these TODOs now, or later?

**Comment by matthchr** [comment]
Closing this in favor of one opened from a branch on this repo rather than my fork

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/controllers/crd_eventgrid_domain_test.go (hunk 2 above)
Comment (matthchr) [reply v2/internal/controllers/crd_eventgrid_domain_test.go:123]: Will do these in a different PR

File: v2/tools/generator/internal/codegen/pipeline/determine_resource_ownership.go (hunk 1 above)
Comment (matthchr) [reply v2/tools/generator/internal/codegen/pipeline/determine_resource_ownership.go:107]: Yes, have fixed this in #2634

**Pull request closed without merging**